File: app/src/main/java/com/example/testlunu/net/acu/cityinfo/Details.java
package com.example.testlunu.net.acu.cityinfo;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class Details {

    @SerializedName("Key")
    @Expose
    public String key;
    @SerializedName("StationCode")
    @Expose
    public String stationCode;
    @SerializedName("StationGmtOffset")
    @Expose
    public Integer stationGmtOffset;
    @SerializedName("BandMap")
    @Expose
    public String bandMap;
    @SerializedName("Climo")
    @Expose
    public String climo;
    @SerializedName("LocalRadar")
    @Expose
    public String localRadar;
    @SerializedName("MediaRegion")
    @Expose
    public Object mediaRegion;
    @SerializedName("Metar")
    @Expose
    public String metar;
    @SerializedName("NXMetro")
    @Expose
    public String nXMetro;
    @SerializedName("NXState")
    @Expose
    public String nXState;
    @SerializedName("Population")
    @Expose
    public Integer population;
    @SerializedName("PrimaryWarningCountyCode")
    @Expose
    public String primaryWarningCountyCode;
    @SerializedName("PrimaryWarningZoneCode")
    @Expose
    public String primaryWarningZoneCode;
    @SerializedName("Satellite")
    @Expose
    public String satellite;
    @SerializedName("Synoptic")
    @Expose
    public String synoptic;
    @SerializedName("MarineStation")
    @Expose
    public String marineStation;
    @SerializedName("MarineStationGMTOffset")
    @Expose
    public Object marineStationGMTOffset;
    @SerializedName("VideoCode")
    @Expose
    public String videoCode;
    @SerializedName("LocationStem")
    @Expose
    public String locationStem;
    @SerializedName("PartnerID")
    @Expose
    public Object partnerID;
    @SerializedName("Sources")
    @Expose
    public List<Source> sources = null;
    @SerializedName("CanonicalPostalCode")
    @Expose
    public String canonicalPostalCode;
    @SerializedName("CanonicalLocationKey")
    @Expose
    public String canonicalLocationKey;

}

File: app/src/main/java/com/example/testlunu/net/acu/cityinfo/Details.kt
package com.example.testlunu.net.acu.cityinfo

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class Details (
    @SerializedName("Key")
    @Expose
    var key: String?,
    @SerializedName("StationCode")
    @Expose
    var stationCode: String?,
    @SerializedName("StationGmtOffset")
    @Expose
    var stationGmtOffset: Int?,
    @SerializedName("BandMap")
    @Expose
    var bandMap: String?,
    @SerializedName("Climo")
    @Expose
    var climo: String?,
    @SerializedName("LocalRadar")
    @Expose
    var localRadar: String?,
    @SerializedName("MediaRegion")
    @Expose
    var mediaRegion: Any?,
    @SerializedName("Metar")
    @Expose
    var metar: String?,
    @SerializedName("NXMetro")
    @Expose
    var nXMetro: String?,
    @SerializedName("NXState")
    @Expose
    var nXState: String?,
    @SerializedName("Population")
    @Expose
    var population: Int?,
    @SerializedName("PrimaryWarningCountyCode")
    @Expose
    var primaryWarningCountyCode: String?,
    @SerializedName("PrimaryWarningZoneCode")
    @Expose
    var primaryWarningZoneCode: String?,
    @SerializedName("Satellite")
    @Expose
    var satellite: String?,
    @SerializedName("Synoptic")
    @Expose
    var synoptic: String?,
    @SerializedName("MarineStation")
    @Expose
    var marineStation: String?,
    @SerializedName("MarineStationGMTOffset")
    @Expose
    var marineStationGMTOffset: Any?,
    @SerializedName("VideoCode")
    @Expose
    var videoCode: String?,
    @SerializedName("LocationStem")
    @Expose
    var locationStem: String?,
    @SerializedName("PartnerID")
    @Expose
    var partnerID: Any?,
    @SerializedName("Sources")
    @Expose
    var sources: List<Source>?,
    @SerializedName("CanonicalPostalCode")
    @Expose
    var canonicalPostalCode: String?,
    @SerializedName("CanonicalLocationKey")
    @Expose
    var canonicalLocationKey: String?
)